internal/sbi/consumer/nrf_service.go

```diff
@@ -0,0 +1,391 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/antihax/optional"
+	"github.com/mohae/deepcopy"
+	"github.com/pkg/errors"
+
+	"github.com/free5gc/openapi"
+	"github.com/free5gc/openapi/Nnrf_NFDiscovery"
+	"github.com/free5gc/openapi/Nnrf_NFManagement"
+	"github.com/free5gc/openapi/models"
+	smf_context "github.com/free5gc/smf/internal/context"
+	"github.com/free5gc/smf/internal/logger"
+)
+
+type nnrfService struct {
+	consumer *Consumer
+
+	NFManagementgMu sync.RWMutex
+	NFDiscoveryMu   sync.RWMutex
+
+	NFManagementClients map[string]*Nnrf_NFManagement.APIClient
+	NFDiscoveryClients  map[string]*Nnrf_NFDiscovery.APIClient
+}
+
+func (s *nnrfService) getNFManagementClient(uri string) *Nnrf_NFManagement.APIClient {
+	if uri == "" {
+		return nil
+	}
+	s.NFManagementgMu.RLock()
+	client, ok := s.NFManagementClients[uri]
+	if ok {
+		defer s.NFManagementgMu.RUnlock()
+		return client
+	}
+
+	configuration := Nnrf_NFManagement.NewConfiguration()
+	configuration.SetBasePath(uri)
+	client = Nnrf_NFManagement.NewAPIClient(configuration)
+
+	s.NFManagementgMu.RUnlock()
+	s.NFManagementgMu.Lock()
+	defer s.NFManagementgMu.Unlock()
+	s.NFManagementClients[uri] = client
+	return client
+}
+
+func (s *nnrfService) getNFDiscoveryClient(uri string) *Nnrf_NFDiscovery.APIClient {
+	if uri == "" {
+		return nil
+	}
+	s.NFDiscoveryMu.RLock()
+	client, ok := s.NFDiscoveryClients[uri]
+	if ok {
+		defer s.NFDiscoveryMu.RUnlock()
```

Comment:
no defer

Reply:
The presence of "defer" does not affect semantics. Is this modification recommended because defer is not strictly necessary?

Reply:
Yes, defer is not ncessary.